internal/ipam/ipam.go

```diff
@@ -1,3 +1,25 @@
 // Package ipam contains types to implement IP address management on IPv4 and
 // IPv6 networks.
 package ipam
+
+import "net"
+
+// Pool is an interface implemented by the IPv4Pool and IPv6Pool types to
+// abstract the type of IP addresses that are managed by the pool.
+type Pool interface {
+	// Obtains the next IP address, or returns nil if the pool was exhausted.
+	GetIP() net.IP
+	// Returns an IP address to the pool. The ip address must have been obtained
+	// by a previous call to GetIP or the method panics.
+	PutIP(net.IP)
```

Comment:
Any reason to stick with `net.IP` rather than use the newer `netip` package?

Reply:
The main reason is the high-level interfaces that are using `net.Addr` (e.g., to get the list of addresses for a network interface). The types in `netip` don't implement `net.Addr` so it results in a lot of type conversions.

That being said, there's still a mix of `net.IP` and `netip.Addr` in the code so I'm going to converge as much of the code to use `netip` 👍